nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java

```diff
@@ -290,6 +291,14 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
 
   @Override
   public void createNamespace(Namespace namespace, Map<String, String> metadata) {
+    if (namespace.levels().length > 1) {
```

Comment:
This would fail, if a parent namespace already exists.

A better place to do this would be `org.apache.iceberg.nessie.NessieIcebergClient#createNamespace`:
* load all required namespaces
* add the `Put` operation for all non-existing parent namespaces to the commit operation

OTOH I'm not sure about the actual contract of this function. Some implementation seem to allow creating a namespaces when parent namespaces do not exist - so the contract isn't clear, aka the functionality per se seems okay to me.

Reply:
@snazy Do you think it makes sense to move this create action to the server side? We also met the same issue when using the REST way to request Nessie. 
I am not cleared with the namespace in the Nessie, not sure if it is a normal request.

Reply:
@zymap we explicitly removed that functionality from the server side a while ago.

Reply:
@snazy I move the impl  to the NessieIcebergClient. PTAL. 

>we explicitly removed that functionality from the server side a while ago.

You mean the Nessie namespace? Could you please point me to the PR?

Reply:
We don't remove namespaces - we have removed the explicit namespace endpoints in Nessie's REST API